compiler/rustc_resolve/src/lib.rs

```diff
@@ -130,6 +130,13 @@ enum Scope<'ra> {
     BuiltinTypes,
 }
 
+/// Scopes used for resolving an `Ident` in a `Module`.
+#[derive(Debug, Copy, Clone, PartialEq)]
+enum ModuleScope {
```

Comment:
Instead of this, I think `enum Scope` above should also get two variants `ModuleNonGlobs` and `ModuleGlobs` instead of one `Module`.

Then finding *any* resolution inside a module would be done using a new `ScopeSet` variant including just two scopes, but other `ScopeSet` would just iterate over `ModuleNonGlobs` and `ModuleGlobs` as a part of the common loop.

This would also be a much larger refactoring.

Reply:
The logic in `early_resolve_ident_in_lexical_scope` should see the explicit resolution and the glob resolution in a module as two separate steps.
So either `Scope::Module` should be split into two scopes, or `resolve_ident_in_module_unadjusted` should somehow produce the both resolutions so that `early_resolve_ident_in_lexical_scope` could process them as if they were from two separate scopes.

Reply:
I'm open to try to implement this, but I still don't understand why this is necessary. Can you explain in more detail, please?

Reply:
Pretty much the same answer as in https://github.com/rust-lang/rust/pull/142547#discussion_r2164582935.
Sorry, I'll try to produce some more details later.